prog/weaver/checkpoint.go

```diff
@@ -1,23 +1,30 @@
 package main
 
 import (
+	"os"
+	"strconv"
 	"sync/atomic"
 	"syscall"
 	"time"
 
-	"github.com/weaveworks/go-checkpoint"
+	checkpoint "github.com/weaveworks/go-checkpoint"
 	weave "github.com/weaveworks/weave/router"
+	api "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
 var checker *checkpoint.Checker
 var newVersion atomic.Value
 var success atomic.Value
 
 const (
-	updateCheckPeriod = 6 * time.Hour
+	updateCheckPeriod  = 6 * time.Hour
+	configMapName      = "weave-net"
+	configMapNamespace = "kube-system"
 )
 
-func checkForUpdates(dockerVersion string, router *weave.NetworkRouter) {
+func checkForUpdates(dockerVersion string, router *weave.NetworkRouter, peers []string) {
```

Comment:
I think I would just pass in `len(peers)`, on the principle of minimum information.

Reply:
Modified.